meta/include/gnuradio-4.0/meta/RangesHelper.hpp

```diff
@@ -59,8 +59,160 @@ struct PairDeduplicateView : std::ranges::range_adaptor_closure<PairDeduplicateV
         }) | std::views::join;
     }
 };
-template<class Eq1, class Eq2>
-PairDeduplicateView(Eq1, Eq2) -> PairDeduplicateView<std::decay_t<Eq1>, std::decay_t<Eq2>>;
+template<class TEq1, class TEq2>
+PairDeduplicateView(TEq1, TEq2) -> PairDeduplicateView<std::decay_t<TEq1>, std::decay_t<TEq2>>;
+
+/**
+ * @brief Lazy k-way merge over a range of sorted ranges.
+ *
+ * Two main components:
+ *  - `MergeView<R, Comp>` - the iterable view that performs the merge lazily.
+ *  - `Merge<Comp>` - the pipeable adaptor that builds a `MergeView` from an outer range.
+ *
+ * Given an outer range `R` whose elements are themselves ranges (the “inner ranges”),
+ * the merge yields a single, sorted sequence by repeatedly selecting the smallest current
+ * element across the inner ranges using `comp`.
+ *
+ * It is required that each inner range is already sorted using `comp`.
+
+ * Usage
+ *  // 1) Pipeable adaptor:
+ *  auto merged = inputs | Merge{compByKey};
+ *  for (auto&& x : merged) { ... }
+ *  // 2) Direct view construction:
+ *   // *  MergeView view{std::views::all(inputs), compByKey};
+ *  for (auto&& x : view) { ... }
+ *
+ * Example
+ *  std::vector<int> a{1,3,5};
+ *  std::vector<int> b{2,4,6};
+ *  std::array ranges{ std::views::all(a), std::views::all(b) };
+ *  auto out = ranges | Merge{std::ranges::less{}};
+ *  // yields: 1,2,3,4,5,6
+ */
+template<class R, class TComp>
+struct MergeView : std::ranges::view_interface<MergeView<R, TComp>> {
+    using TOut      = R;
+    using TInView   = std::remove_cvref_t<std::ranges::range_reference_t<TOut>>;
+    using TIterator = std::ranges::iterator_t<TInView>;
+    using TSentinel = std::ranges::sentinel_t<TInView>;
+
+    static_assert(std::ranges::forward_range<TInView>);
+
+    TOut                        _out;
+    [[no_unique_address]] TComp _comp{};
+
+    MergeView() = default;
+
+    template<class TRFw, class TCompFw>
+    requires std::constructible_from<TOut, TRFw> && std::constructible_from<TComp, TCompFw>
+    constexpr MergeView(TRFw&& r, TCompFw&& c) : _out(std::forward<TRFw>(r)), _comp(std::forward<TCompFw>(c)) {}
+
+    struct Iterator {
+        using iterator_concept  = std::forward_iterator_tag;
+        using iterator_category = std::forward_iterator_tag;
+        using difference_type   = std::ptrdiff_t;
+        using reference         = std::iter_reference_t<TIterator>;
+        using value_type        = std::iter_value_t<TIterator>;
+
+        MergeView*             _view{};
+        std::vector<TIterator> _its;
+        std::vector<TSentinel> _ends;
```

Comment:
```suggestion
        std::vector<TIterator> _its;
```
I think some of the changes we discussed somehow didn't get pushed properly, the `_ends` vector should not be necessary anymore.

Note: same as above, this allocated stoage is only strictly required for blocks with `vector<Port>`, since the number of ports and inputranges is not known at compile time here. For all other blocks an implementation with fixed storage will be added.